### PR TITLE
refactor: summarize vault type badges

### PR DESCRIPTION
**Pull request by Seranged**
## Summary
- Pair information shows vault type summaries only when a side has a private, cyclical note, or unknown badge.
- Shared vault type badge derivation lives in one composable so full badge rendering and summary rendering stay aligned.

## Changes
- Added `useVaultTypeBadges` to derive governance, private, governance-limited, securitize, and cyclical badges from the same source.
- Added `summaryOnly` rendering to `VaultTypeBadges` for pair summaries.
- Updated `VaultOverviewPairBlockTypes` to hide the section when neither side has summary badges.
- Added focused tests for private, unknown, cyclical IRM types 4/5, KINKY IRM type 3, and hidden normal/governance-limited cases.

## Test plan
- [x] `npm run test:run -- tests/composables/useVaultTypeBadges.test.ts tests/entities/vault/utils.test.ts`
- [x] `npm run typecheck`
- [x] `npx eslint composables/useVaultTypeBadges.ts components/entities/vault/VaultTypeBadges.vue components/entities/vault/overview/VaultOverviewPairBlockTypes.vue tests/composables/useVaultTypeBadges.test.ts`
- [x] Manually checked a private pair renders `Private` in pair information.
- [x] Manually checked a normal governed pair hides the vault types section.

## Summary by CodeRabbit

* **New Features**
  * Added a compact badge summary mode to show only key vault indicators (cyclical note, private, unknown) where appropriate; overview displays now opt into this summary mode.

* **Refactor**
  * Centralized badge and governance-type logic into a shared utility for consistent badge visibility and simpler templates.

* **Tests**
  * Added tests covering badge combinations, verification, governance states, and summary behavior.

[Review Change Stack](https://app.coderabbit.ai/change-stack/euler-xyz/euler-lite/pull/488?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: components/entities/vault/VaultTypeBadges.vue [hunk 1]
```diff
@@ -1,69 +1,23 @@
...
-  return isCyclicalNoteVault(vault as Vault)
-})
+const vaultRef = computed(() => vault)
+const { badges, governanceType, summaryBadges } = useVaultTypeBadges(vaultRef)
 
 const isStacked = computed(() => layout === 'stacked')
 const tagElement = computed(() => isStacked.value ? 'button' : 'span')
+const visibleBadges = computed(() => summaryOnly ? summaryBadges.value : badges.value)
+const hasVisibleBadge = (badge: VaultTypeBadge): boolean => visibleBadges.value.includes(badge)
+const showGovernanceType = computed(() => hasVisibleBadge(governanceType.value))
```
Comment (LeonardEulerXYZ) [inline components/entities/vault/VaultTypeBadges.vue]: This condition drops one of the summary-mode cases. `summaryBadges` includes `'unknown'` whenever `!isVerified.value`, but `governanceType` can still be `'governed'`/`'managed'` if the governor/owner address is present in label entities while the verification rule fails (e.g. `vault.verified === false`, product entity mismatch, or router-governor mismatch). In that case `hasSummaryBadges` makes the pair block render, but `showGovernanceType` is false and no `Unknown` chip is shown for that side.

Consider rendering the governance chip when `summaryOnly && summaryBadges.includes('unknown')`, and pass an effective type of `'unknown'` (or otherwise align `visibleBadges` with the type checked here). A small regression test with an unverified vault whose `governorAdmin` is in `getEntitiesByVault()` would pin the invariant.

**Review by LeonardEulerXYZ — CHANGES_REQUESTED** [review]
Reviewed PR #488 at head 69e402ab23753355d9f2c4fa51507c2e072ebcb2.

Verdict: Request changes — one summary-mode edge case can hide the Unknown badge while still showing the Vault types summary block.

Validation performed:
- `npm run test:run -- tests/composables/useVaultTypeBadges.test.ts tests/entities/vault/utils.test.ts` passed
- `npm run typecheck` passed
- `npx eslint composables/useVaultTypeBadges.ts components/entities/vault/VaultTypeBadges.vue components/entities/vault/overview/VaultOverviewPairBlockTypes.vue tests/composables/useVaultTypeBadges.test.ts` passed
- `npm run build` passed (only existing Sentry/Reown/Rollup chunk warnings)

What looks good:
- The badge derivation is nicely centralized in `useVaultTypeBadges`.
- The focused tests cover ordinary governed, private, cyclical, KINKY, unknown, and governance-limited cases.

**Review by kasperpawlowski — DISMISSED** [review]

**Review by LeonardEulerXYZ — COMMENTED** [review]
Re-reviewed latest head `070f0efd20fb15c5a4e417b179f4730de58af189`.

The previous blocker is resolved: summary rendering now uses `summaryGovernanceType`, so a vault that fails verification renders `Unknown` in summary mode even when the raw governance classification is `governed`/`managed`. The added regression test covers this exact case.

Validation run locally on a clean worktree:
- `npm run test:run -- tests/composables/useVaultTypeBadges.test.ts` — passed, 8/8 tests
- `npm run typecheck` — passed
- `npx eslint components/entities/vault/VaultTypeBadges.vue components/entities/vault/overview/VaultOverviewPairBlockTypes.vue composables/useVaultTypeBadges.ts tests/composables/useVaultTypeBadges.test.ts` — passed
- `npm run build` — passed

Notes:
- Build warnings are the existing Sentry auth/source-map and Rollup chunk/dynamic-import warnings; I did not see a PR-specific blocker there.
- GitHub checks currently show CodeRabbit skipped/pass and Railway dev-build pass.

No remaining blocking findings from this re-review.

**Review by kasperpawlowski — APPROVED** [review]
